plugins/CircuitViewer/tests/braynsTestData.cpp

```diff
@@ -113,11 +117,51 @@ BOOST_AUTO_TEST_CASE(circuit_with_simulation_mapping)
 
     modelDesc->getModel().getTransferFunction().setValuesRange({-66, -62});
 
-    brayns.commit();
+    clientServer.getBrayns().commit();
     modelDesc->getModel().getSimulationHandler()->waitReady();
-    brayns.commitAndRender();
-    BOOST_CHECK(compareTestImage("testdataallmini50basicsimulation.png",
-                                 brayns.getEngine().getFrameBuffer()));
+    clientServer.getBrayns().commitAndRender();
+    BOOST_CHECK(compareTestImage(
+        "testdataallmini50basicsimulation.png",
+        clientServer.getBrayns().getEngine().getFrameBuffer()));
+
+    brayns::SnapshotParams params;
+    params.format = "png";
+    params.samplesPerPixel = 16;
+    params.size = clientServer.getBrayns()
+                      .getParametersManager()
+                      .getApplicationParameters()
+                      .getWindowSize();
+    params.animParams = std::make_unique<brayns::AnimationParameters>(
+        clientServer.getBrayns()
+            .getParametersManager()
+            .getAnimationParameters());
+    params.animParams->setFrame(42);
+
+    auto image = clientServer.makeRequest<
+        brayns::SnapshotParams, brayns::ImageGenerator::ImageBase64>("snapshot",
+                                                                     params);
+    auto decodedImage = base64_decode(image.data);
+    const auto referenceFilename = BRAYNS_TESTDATA_IMAGES_PATH
+        "testdataallmini50basicsimulation_snapshot.png";
+#ifdef GENERATE_REFERENCE_SNAPSHOT
+    {
+        std::fstream file(referenceFilename, std::ios::out);
+        file << decodedImage;
+    }
+#endif
+
+    const std::string newFilename(
+        (fs::temp_directory_path() / fs::unique_path()).string());
+    {
+        std::fstream file(newFilename, std::ios::out);
+        file << decodedImage;
+    }
+    const auto newImage{pdiff::read_from_file(newFilename)};
+    const auto referenceImage{pdiff::read_from_file(referenceFilename)};
+    const auto liveImage{pdiff::read_from_file(
+        BRAYNS_TESTDATA_IMAGES_PATH "testdataallmini50basicsimulation.png")};
+    BOOST_CHECK(pdiff::yee_compare(*referenceImage, *newImage));
+    BOOST_CHECK(!pdiff::yee_compare(*liveImage, *newImage));
```

Comment:
Why don't you try to refactor some of this code to a compareTestImage function that could share code with the function in PDiffHelpers.h?

Reply:
Because that code was first (see snapshot.cpp), so once someone did compareTestImage did not unify in the first place.

Reply:
Are you implying that it should be @karjonas who does it?

Reply:
I see, I didn't understand it that way. It's true that if you take the code from snapshot, you're seemingly doing the right thing.

But are you going to give it a try or leave it for another PR?

Reply:
I can try, yes.